components/DashboardPage/components/Pomodoro/Timer.tsx

```diff
@@ -0,0 +1,103 @@
+import moment from 'moment';
+import { useState, useEffect } from 'react';
+import { Stack, Typography, Button } from 'components';
+
+interface TimerProps {
+  round: number;
+  setRound: any;
+}
+
+export function Timer({ round, setRound }: TimerProps) {
+  let workTime = 25 * 60;
+  let breakTime = 5 * 60;
+  const [time, setTime] = useState<number>(workTime);
+  const [clock, setClock] = useState<boolean>(true);
+  const [countdown, setCountdown] =
+    useState<boolean>(false);
+  const [workState, setWorkState] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    if (time === 0) {
+      setWorkState((workState) => !workState);
+      if (workState === false) {
+        setTime(breakTime);
+        setRound((round: number) => round + 1);
+      }
+      if (workState === true) setTime(workTime);
+    }
+
+    if (countdown && time > 0) {
+      setTimeout(() => {
+        setClock((clock) => !clock);
+      }, 1000);
+      setTime(time - 1);
+    }
+  }, [countdown, clock, workState]);
+
+  var momentDurationFormatSetup = require('moment-duration-format');
+
+  const formattedTimeLeft = moment
+    .duration(time, 's')
+    .format('mm:ss', { trim: false });
+
+  return (
+    <Stack
+      center
+      round
+      background="gradient"
+      width="230px"
+      height="230px"
+      justifyCenter
+    >
+      <Stack
+        vertical
+        center
+        round
+        background="widgetBackgroundColor"
+        width="210px"
+        height="210px"
+        py="20px"
+        spacing="tight"
+      >
+        {workState ? (
+          <Stack
+            background="mainBackgroundColor"
+            py="tight"
+            px="tight"
+            opacity="0.7"
+          >
+            <Typography type="smallText" uppercase bold>
+              Break
+            </Typography>
+          </Stack>
+        ) : (
+          <Stack
+            background="mainBackgroundColor"
+            py="tight"
+            px="tight"
+            opacity="0.7"
+          >
+            <Typography type="smallText" uppercase bold>
+              Work
+            </Typography>
+          </Stack>
+        )}
+        <Typography type="header1" thin>
+          {formattedTimeLeft}
+        </Typography>
+        <Button
+          label={countdown ? 'Stop' : 'Start'}
+          variant="nav"
+          onClick={startCountdown}
+        />
+      </Stack>
+    </Stack>
+  );
+
+  function startCountdown() {
+    if (time > 0) {
+      setCountdown(!countdown);
+    }
+  }
+}
```

Comment:
I noticed that when you start the timer for the very first round, the PomodoroBoard remains at 0 for that Round and the Goal. Is that an expected behavior? If so, disregard this comment 👍 . I feel like a Round 0 and goal 0 doesn't really make sense so if it's indeed an undesired behavior, I think you could set the round for the special case of round === 0 in this function.

Reply:
It starts at 0 because you complete one round only after the first 25 minutes :D